pyomo/dae/tests/test_simulator.py

```diff
@@ -20,18 +21,18 @@
     Simulator, 
     _check_getitemexpression, 
     _check_productexpression,
-    _check_sumexpression, 
-    substitute_getitem_with_casadi_sym,
-    substitute_intrinsic_function_with_casadi,
-    substitute_intrinsic_function)
-from pyomo.core.base import expr as EXPR
-from pyomo.core.base import expr_common
+    _check_viewsumexpression, 
+    substitute_pyomo2casadi,
+    #substitute_getitem_with_casadi_sym,
+    #substitute_intrinsic_function_with_casadi,
+    #substitute_intrinsic_function)
+)
 from pyomo.core.base.template_expr import (
     IndexTemplate, 
     _GetItemIndexer,
-    substitute_template_expression, 
-    substitute_getitem_with_param,
-    substitute_template_with_value,
+    #substitute_template_expression, 
+    #substitute_getitem_with_param,
+    #substitute_template_with_value,
```

Comment:
These commented imports should be removed. Also, I think you added an extra expression checker to the Simulator for NegationExpressions. We should add a test for that, I'll commit one shortly.